code/backend/batch/utilities/search/SearchHandlerBase.py

```diff
@@ -1,13 +1,24 @@
 from abc import ABC, abstractmethod
+from ..helpers.EnvHelper import EnvHelper
 
 from ..common.SourceDocument import SourceDocument
 
 
 class SearchHandlerBase(ABC):
-    def __init__(self, env_helper):
+    def __init__(self, env_helper: EnvHelper):
         self.env_helper = env_helper
         self.search_client = self.create_search_client()
 
+    def search_with_facets(self, query, facets):
```

Comment:
can we put some types on these as it looks like it is a dictionary?

Reply:
Updated